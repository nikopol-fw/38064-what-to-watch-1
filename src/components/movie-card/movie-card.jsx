// movie-card.jsx

import React from 'react';
import PropTypes from 'prop-types';


export const MovieCard = (props) => {
  const {title, onClick} = props;

  return <article className="small-movie-card catalog__movies-card">
    <button className="small-movie-card__play-btn" type="button">Play</button>
    <div className="small-movie-card__image">
      <img src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg" alt="{title}" width="280" height="175" />
    </div>
    <h3 className="small-movie-card__title">
      <a className="small-movie-card__link"
        href="movie-page.html"
        onClick={onClick}>{title}</a>
    </h3>
  </article>;
};


MovieCard.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func
};
