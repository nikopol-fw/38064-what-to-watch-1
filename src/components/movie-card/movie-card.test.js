// movie-card.test.jsx

import React from 'react';
import renderer from 'react-test-renderer';

import {MovieCard} from './movie-card';


it(`MovieCard correctly renders`, () => {
  const tree = renderer
    .create(<MovieCard
      title={`Test title`}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
