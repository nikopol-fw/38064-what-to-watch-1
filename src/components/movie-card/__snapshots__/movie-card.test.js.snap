// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard correctly renders 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height={175}
      loop={true}
      muted={true}
      poster="/img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      preload="metadata"
      width={280}
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
